rasa/core/policies/ted_policy.py

```diff
@@ -761,7 +1397,131 @@ def _process_batch_data(
 
         batch_features = tf.concat(batch_features, axis=-1)
 
-        return batch_features
+        return batch_features, text_transformer_output, text_sequence_lengths
+
+    def _reshape_for_entities(
+        self,
+        tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]],
+        dialogue_transformer_output: tf.Tensor,
+        text_transformer_output: tf.Tensor,
+        text_sequence_lengths: tf.Tensor,
+    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]:
+        # To calculate the loss for entities we need the output of the text
+        # sequence transformer (shape: real entity dim x
```

Comment:
Really helpful comment block. What is `real entity dimension`?

Reply:
Some where in the code above comments, we have a lengthy description of what "real" vs "fake" features are

Reply:
I see your point, I rephrased